torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -82,6 +82,9 @@ LTCTensorImpl::LTCTensorImpl(LazyTensor&& tensor)
   // This is a temporary fix for a PyTorch core issue,
   // according to https://github.com/pytorch/xla/pull/2682.
   is_non_overlapping_and_dense_ = false;
+
+  // Error out on sizes() if PyTorch is built w/ C10_DISABLE_TENSORIMPL_EXTENSIBILITY
+  set_sizes_customization_policy(CustomizableMethodPolicy::NotSupported);
```

Comment:
cool! how did you test this?

Reply:
I'm afraid you'll ask that. I copied this approach from NestedTensors which similarly overrides `sizes()` but doesn't support extensibility. If we run LTC tests in fbcode I can add a test specifically if extensibility is on and assert there was an exception or we just hope `NestedTensor` folks knew what they are doing :)

Reply:
i was kind of hesitating saying anything bc i only wanted to have you add a test if it is easy and i'm not sure if it's easy.  we do run limited parts of ltc tests in fbcode, basically any part that is not guarded by #ifdef, 
e.g. we skip ops tests:
https://github.com/pytorch/pytorch/blob/master/test/cpp/lazy/test_lazy_ops.cpp#L18

but we run misc tests:
https://github.com/pytorch/pytorch/blob/master/test/cpp/lazy/test_misc.cpp#L23

Reply:
Sounds gut, let me give it a try!